tests/ha/single_pod_restart/test_server_pod_restart.py

```diff
@@ -717,3 +718,171 @@ def test_reads_during_server_pod_restart(self):
+        thread_rd.daemon = True  # Daemonize thread
+        LOGGER.info("Step 2.2: Successfully started READs and verify DI on the written data in "
+                    "background")
+
+        LOGGER.info("Step 2.3: Starting DELETEs of %s buckets in background", del_bucket)
+        del_buckets = written_bck.copy()
+        args = {'test_prefix': test_prefix_del, 'test_dir_path': self.test_dir_path,
+                'skipput': True, 'skipget': True, 'bkt_list': del_buckets,
+                'bkts_to_del': del_bucket, 'output': del_output}
+        thread_del = threading.Thread(target=self.ha_obj.put_get_delete,
+                                      args=(event, s3_test_obj,), kwargs=args)
+        thread_del.daemon = True  # Daemonize thread
+        thread_wri.start()
+        thread_rd.start()
+        thread_del.start()
+        LOGGER.info("Step 2.3: Successfully started DELETEs of %s buckets in background",
+                    del_bucket)
+        LOGGER.info("Step 2: Started WRITEs, READs and DELETEs with variable object sizes "
+                    "during server pod restart using kubectl delete")
+        LOGGER.info("Waiting for %s seconds", HA_CFG["common_params"]["10sec_delay"])
+        time.sleep(HA_CFG["common_params"]["10sec_delay"])
+
+        LOGGER.info("Step 3: Restart the server pod by kubectl delete.")
+        LOGGER.info("Get pod name to be deleted")
+        pod_list = self.node_master_list[0].get_all_pods(pod_prefix=const.SERVER_POD_NAME_PREFIX)
+        pod_name = self.system_random.sample(pod_list, 1)[0]
+        LOGGER.info("Deleting pod %s", pod_name)
+        event.set()
+        resp = self.node_master_list[0].delete_pod(pod_name=pod_name, force=True)
+        LOGGER.debug("Response: %s", resp)
+        assert_utils.assert_true(resp[0], f"Failed to delete pod {pod_name} by kubectl delete")
+        LOGGER.info("Step 3: Successfully restarted pod %s by kubectl delete", pod_name)
+
+        LOGGER.info("Step 4: Check cluster status")
+        resp = self.ha_obj.poll_cluster_status(self.node_master_list[0])
+        assert_utils.assert_true(resp[0], resp)
+        LOGGER.info("Step 4: Cluster is in good state")
+        event.clear()
+        thread_wri.join()
+        thread_rd.join()
+        thread_del.join()
+
+        LOGGER.info("Background WRITEs, READs and DELETEs threads joined successfully.")
+        LOGGER.info("Step 5: Verify responses from background processes")
+        LOGGER.info("Step 5.1: Verify status for In-flight WRITEs")
+        responses_wr = dict()
+        while len(responses_wr) != 2:
+            responses_wr = output_wr.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        pass_logs = list(x[1] for x in responses_wr["pass_res"])
+        fail_logs = list(x[1] for x in responses_wr["fail_res"])
+        resp = self.ha_obj.check_s3bench_log(file_paths=pass_logs)
+        assert_utils.assert_false(len(resp[1]), f"WRITEs logs which contain failures: {resp[1]}")
```

Comment:
maybe we can enhance this error message saying Writes not expected to fail before and after pod restarted, same for others

Reply:
Yes, message can be enhanced. We will have to do this at all the places. Because almost at all places we are using same error message.